File: bongf/python/level1/harshad_number.py
def solution(x):
    return True if (x % sum( int(i) for i in iter(str(x)))) == 0 else False

# string을 for문 돌리면 ch가 나와서 iter 필요 없고 
# 저번에 했던 것과 같이 if else 써줄 필요 없다. 

def solution2(n):
    return n % sum([int(c) for c in str(n)]) == 0

print(solution(10))
print(solution(12))
print(solution(11))
print(solution(13))

File: programmers/java/src/level1/HarshadNumber.java
package level1;

import java.util.stream.Stream;

public class HarshadNumber {
    public static void main(String[] args) {
        System.out.println(solution(10));
        System.out.println(solution(12));
        System.out.println(solution(11));
        System.out.println(solution(13));

    }

    public static boolean solution(int x) {
        return x % Stream.of(String.valueOf(x)
                .split("")).map(Integer::valueOf)
                .reduce(0, Integer::sum) == 0;
    }
}
